src/components/Switch/Switch.tsx

```diff
@@ -1,184 +1,231 @@
+                        width: toggleSize,
+                        height: toggleSize,
+                        borderRadius: (size * 0.84) / 2,
+                    },
+                    thumbColor && { backgroundColor: thumbColor },
+                    !value &&
+                        !unCheckedIcon &&
+                        !pressed && {
+                            borderRadius: (size * 0.5) / 2,
+                        },
+                    pressed && {
+                        borderRadius: (size * (!value ? 1 : 0.85)) / 2,
+                    },
+                ],
+                overlayStyle: [
+                    toggleWheel,
+                    {
+                        marginLeft: moveToggle,
+                    },
+                    thumbOverlay,
+                ],
+                iconStyle: icon,
+            };
+        }, [
+            componentStyles,
+            size,
+            color,
+            moveToggle,
+            toggleSize,
+            thumbColor,
+            value,
+            unCheckedIcon,
+            pressed,
+            thumbOverlay,
+        ]);
 
-export const defaultStyles: ComponentStylePropWithVariants<
-    ViewStyle,
-    'selected_disabled' | 'active' | 'disabled',
-    CustomProp
-> = {
-    thumbTintColor: 'colors.neutral1',
-    onTintColor: 'colors.onNeutral1',
+    const handleValueChange = useCallback(() => {
+        onValueChange && onValueChange(!value);
+    }, [value, onValueChange]);
 
-    states: {
-        active: {
-            thumbTintColor: 'colors.neutral1',
-            onTintColor: 'colors.onNeutral1',
+    const renderIcon = useCallback(
+        (icon: string | undefined, condition: boolean, type?: IconType) => {
+            return icon && condition ? (
+                <Icon style={iconStyle} name={icon} size={size * 0.5} type={type} />
+            ) : null;
         },
-        disabled: {
-            thumbTintColor: 'colors.disabled',
-            onTintColor: 'colors.disabledOnBackground',
-        },
-        selected_disabled: {
-            thumbTintColor: 'colors.neutral1',
-            onTintColor: 'colors.disabledOnBackground',
-        },
-    },
+        [Icon, iconStyle, size],
+    );
+
+    const onIconComponent = useMemo(
+        () => renderIcon(checkedIcon, !!value, checkedIconType),
+        [renderIcon, checkedIcon, value, checkedIconType],
+    );
+    const offIconComponent = useMemo(
+        () => renderIcon(unCheckedIcon, !value, uncheckedIconType),
+        [renderIcon, unCheckedIcon, value, uncheckedIconType],
+    );
+
+    return (
+        <Pressable ref={actionsRef} style={buttonStyle} onPress={handleValueChange}>
+            <View style={toggleStyle}>
+                <Animated.View style={parentViewStyle as ViewStyle}>
+                    <Animated.View style={toggleWheelStyle}>
+                        {onIconComponent}
+                        {!disabled && offIconComponent}
```

Comment:
this look misleading. Seem like this is saying if switch is enabled render offIconComponent.
We need to combine the two of them into one.

Reply:
And why don't we want to show the offIcon if switch is disabled?

Reply:
I think it was in the old M3 guidelines maybe thats why I did it like that, I removed this condition now. @thetaungg